### PR TITLE
python@3.10 3.10.2

### PR DESCRIPTION
**Pull request by branchv**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

---

### CODE REVIEW

**Comment by carlocab** [comment]
Let's not run the tests while Rust and Boost are still running.

**Comment by cclauss** [comment]
Gentle bump please.  Rust and Boost both seem to be blocked.

**Comment by branchvincent** [comment]
Thanks, re-running now since the CI queue is clear

File: Formula/python@3.10.rb [hunk 1]
```diff
@@ -45,8 +45,8 @@ class PythonAT310 < Formula
               "bin/easy_install-3.7", "bin/easy_install-3.8", "bin/easy_install-3.9"
 
   resource "setuptools" do
-    url "https://files.pythonhosted.org/packages/e6/e2/f2bfdf364e016f7a464db709ea40d1101c4c5a463dd7019dae0a42dbd1c6/setuptools-59.5.0.tar.gz"
-    sha256 "d144f85102f999444d06f9c0e8c737fd0194f10f2f7e5fdb77573f6e2fa4fad0"
+    url "https://files.pythonhosted.org/packages/69/77/aee1ecacea4d0db740046ce1785e81d16c4b1755af50eceac4ca1a1f8bfd/setuptools-60.5.0.tar.gz"
```
Comment (Bo98) [inline Formula/python@3.10.rb:48]: Per an agreement I made with upstream, this contains breaking changes which require an additional patch from our side for sysconfig (the one I mentioned on Slack before the holidays). I'll add it here tonight.

Comment (carlocab) [reply Formula/python@3.10.rb:48]: Killing CI for now.

**Comment by Bo98** [comment]
Here's the patches I had before: https://github.com/Bo98/formula-patches/commit/3306d3e87bc042de2381202f4f508fdafbc075d3

But looking at it again I wonder if it breaks edge cases like `pip install --prefix /custom/prefix`. I'll need some time to test all this but I'll report back before the end of the week.

**Comment by cclauss** [comment]
Gentle bump here...  There is a memory leak fixed in 3.10.1 that this release fixes.
https://docs.python.org/release/3.10.2/whatsnew/changelog.html

**Comment by Bo98** [comment]
Yeah I’m pushing something tonight

**Comment by carlocab** [comment]
We probably need this:

```diff
diff --git a/Formula/python@3.10.rb b/Formula/python@3.10.rb
index 02a84f15932..a2b2f2dc758 100644
--- a/Formula/python@3.10.rb
+++ b/Formula/python@3.10.rb
@@ -193,7 +193,7 @@ class PythonAT310 < Formula
       # Prevent third-party packages from building against fragile Cellar paths
       inreplace Dir[lib_cellar/"**/_sysconfigdata__darwin_darwin.py",
                     lib_cellar/"config*/Makefile",
-                    frameworks/"Python.framework/Versions/3*/lib/pkgconfig/python-3.?.pc"],
+                    frameworks/"Python.framework/Versions/3*/lib/pkgconfig/python-3*.pc"],
                 prefix, opt_prefix
 
       # Help third-party packages find the Python framework
@@ -210,7 +210,7 @@ class PythonAT310 < Formula
       inreplace Dir[lib_cellar/"**/_sysconfigdata_*linux_x86_64-*.py",
                     lib_cellar/"config*/Makefile",
                     bin/"python#{version.major_minor}-config",
-                    lib/"pkgconfig/python-3.?.pc"],
+                    lib/"pkgconfig/python-3*.pc"],
                 prefix, opt_prefix
 
       inreplace bin/"python#{version.major_minor}-config",
```
The existing glob assumes a single-digit minor version (and also excludes some other `.pc` files even if it accounted for a double-digit minor version).

**Comment by carlocab** [comment]
On all macOS:
```
Error: 2 failed steps!
brew test --retry --verbose condure
brew test --retry --verbose zurl
```

Judging from the `brew test` output, `condure` and `zurl` are the same issue, and [`zurl` is pre-existing](https://github.com/Homebrew/homebrew-core/pull/92549).

On Linux:
```
Error: 3 failed steps!
brew test --retry --verbose iproute2mac
brew linkage --test opencascade
brew test --retry --verbose vim
```

`iproute2mac` should be macOS-only. `opencascade` is pre-existing and unrelated. I don't know what to make of `vim`. Can someone take a look here, @Homebrew/linux?

**Review by carlocab — APPROVED** [review]
